mne_connectivity/spectral/epochs_multivariate.py

```diff
@@ -99,35 +100,51 @@ class _EpochMeanMultivariateConEstBase(_AbstractConEstBase):
 
     name = None
     n_steps = None
+    con_scores = None
     patterns = None
     filters = None
     con_scores_dtype = np.float64
 
     def __init__(
-        self, n_signals, n_cons, n_freqs, n_times, *, store_filters=False, n_jobs=1
+        self,
+        n_signals,
+        n_cons,
+        n_freqs,
+        n_times,
+        *,
+        n_components=0,
+        store_con=True,
+        store_filters=False,
+        n_jobs=1,
     ):
         self.n_signals = n_signals
         self.n_cons = n_cons
         self.n_freqs = n_freqs
         self.n_times = n_times
+        self.n_components = n_components
+        self.store_con = store_con
         self.store_filters = store_filters
         self.n_jobs = n_jobs
 
-        # include time dimension, even when unused for indexing flexibility
+        # allocate space for accumulation of CSD
+        csd_shape = (n_signals**2, n_freqs, 1 if n_times == 0 else n_times)
+        self._acc = np.zeros(csd_shape, dtype=np.complex128)
         if n_times == 0:
-            self.csd_shape = (n_signals**2, n_freqs)
-            self.con_scores = np.zeros(
-                (n_cons, n_freqs, 1), dtype=self.con_scores_dtype
-            )
-        else:
-            self.csd_shape = (n_signals**2, n_freqs, n_times)
+            self._acc = np.squeeze(self._acc, axis=-1)
+
+        # allocate space for storing results
+        # include time & components dimensions for indexing flexibility, even if unused
+        if store_con:
             self.con_scores = np.zeros(
-                (n_cons, n_freqs, n_times), dtype=self.con_scores_dtype
+                (
+                    n_cons,
+                    1 if n_components == 0 else n_components,
+                    n_freqs,
+                    1 if n_times == 0 else n_times,
+                ),
+                dtype=self.con_scores_dtype,
             )
```

Comment:
Similar to how `n_times=0` means there is no temporal element to the results but this dimension is still included for indexing flexibility, `n_components=0` acts in the same way.